core/src/main/java/com/scalar/db/util/ProtoUtils.java

```diff
@@ -95,41 +99,52 @@ private static Key toKey(com.scalar.db.rpc.Key key) {
         if (value.getTextValue().hasValue()) {
           return new TextValue(value.getName(), value.getTextValue().getValue());
         } else {
-          return new TextValue(value.getName(), (String) null);
+          return null;
         }
       case BLOB_VALUE:
         if (value.getBlobValue().hasValue()) {
           return new BlobValue(value.getName(), value.getBlobValue().getValue().toByteArray());
         } else {
-          return new BlobValue(value.getName(), (byte[]) null);
+          return null;
         }
+      case VALUE_NOT_SET:
```

Comment:
For the protocol buffer, if the value is not set (`VALUE_NOT_SET`), we tread it as a NULL value.